app/models/shipit/deploy.rb

```diff
@@ -97,14 +97,16 @@ def build_rollback(user = nil, env: nil, force: false)
     end
 
     # Rolls the stack back to this deploy
-    def trigger_rollback(user = AnonymousUser.new, env: nil, force: false)
+    def trigger_rollback(user = AnonymousUser.new, env: nil, force: false, lock: true)
       rollback = build_rollback(user, env: env, force: force)
       rollback.save!
       rollback.enqueue
 
-      lock_reason = "A rollback for #{rollback.since_commit.sha} has been triggered. " \
-        "Please make sure the reason for the rollback has been addressed before deploying again."
-      stack.update!(lock_reason: lock_reason, lock_author_id: user.id)
+      if lock
+        lock_reason = "A rollback for #{rollback.since_commit.sha} has been triggered. " \
+          "Please make sure the reason for the rollback has been addressed before deploying again."
+        stack.update!(lock_reason: lock_reason, lock_author_id: user.id)
```

Comment:
Do we need to trigger a stack update in a different way now, to ensure any related hooks will still be processed when a rollback is triggered? Or is creating and enqueueing the rollback sufficient?

That is, could we have been relying on this update! call for other side-effects?

Reply:
Looking at the events on update I don't believe we need those since the stack isn't actually changing, but I'm not super familiar with shipit-engine so I'm open to being wrong. When I went through them everything related to the stack actually changing and broadcasting that change which is doesn't in this case (and we don't trigger them on a regular deploy either from what I can see).

FWIW I was able to run this locally and didn't hit any roadblocks, but if there's extra stuff you think I should test that I might not know about, I'm super game!